test/sdiff.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import json
+
+def sdiff(actual, expected, path):
+  try:
+    if expected is None and (
+        actual == 0 or actual == "" or actual == False or \
+        actual is None or actual == [] or actual == {}
+    ):  # equate default values to None
+      return (True, "")
+
+    if type(actual) != type(expected):
+      return (False, path)
+
+    if isinstance(actual, list):
+      if len(actual) != len(expected):
+        return (False, path)
+
+      for i in range(len(actual)):
+        status, rpath = sdiff(actual[i], expected[i], "%s/%d" % (path, i))
+        if not status:
+          return (False, rpath)
```

Comment:
Maybe replace False with status? Makes it clearer that you're just passing on the result.